spec/01-unit/006-schema_validation_spec.lua

```diff
@@ -292,6 +292,22 @@ describe("Schemas", function()
         assert.falsy(err)
         assert.same({"hello", "world"}, values.array)
       end)
+
+      it("should handle escaped commas in comma-separated arrays", function()
+        local s = {
+          fields = {
+            array = {type = "array"}
+          }
+        }
+
+        -- It should also strip the resulting strings
+        local values = { array = "hello\\, world,goodbye world\\,,/hello/\\d{1\\,3}" }
+
+        local valid, err = validate_entity(values, s)
+        assert.True(valid)
+        assert.falsy(err)
+        assert.same({"hello, world", "goodbye world,", "/hello/\\d{1,3}"}, values.array)
```

Comment:
Would you agree that this test could be made more readable? It also says it will strip the resulting strings, but is not doing so. For the sake of the tests' atomicity, it might be a good thing though!

Here is a proposed update that just makes the test more readable (avoids double escaping because of Lua strings) (the rest is just more updated style/test message that is - granted - not followed by the rest of this outdated test suite...):

```diff
diff --git a/spec/01-unit/006-schema_validation_spec.lua b/spec/01-unit/006-schema_validation_spec.lua
index cc1ca8d7..8f2f8156 100644
--- a/spec/01-unit/006-schema_validation_spec.lua
+++ b/spec/01-unit/006-schema_validation_spec.lua
@@ -293,20 +293,26 @@ describe("Schemas", function()
         assert.same({"hello", "world"}, values.array)
       end)
 
-      it("should handle escaped commas in comma-separated arrays", function()
+      it("preserves escaped commas in comma-separated arrays", function()
+        -- Note: regression test for arrays of PCRE URIs:
+        -- https://github.com/Mashape/kong/issues/2780
+
         local s = {
           fields = {
-            array = {type = "array"}
+            array = { type = "array" }
           }
         }
 
         -- It should also strip the resulting strings
-        local values = { array = "hello\\, world,goodbye world\\,,/hello/\\d{1\\,3}" }
+        local values = {
+          array = [[  hello\, world  ,  goodbye world  ,  /hello/\d{1\,3}  ]]
+        }
 
         local valid, err = validate_entity(values, s)
         assert.True(valid)
         assert.falsy(err)
-        assert.same({"hello, world", "goodbye world,", "/hello/\\d{1,3}"}, values.array)
+        assert.same({ "hello, world", "goodbye world", [[/hello/\d{1,3}]] },
+                    values.array)
       end)
     end)
   end)
```